### PR TITLE
CB-27090 - Produce dedicated log files for secret encryption scripts

### PR DESCRIPTION
**Pull request by gregito**

### CODE REVIEW

**Review by lajosrodek — CHANGES_REQUESTED** [review]
Two minor asks.

File: saltstack/base/salt/luks/etc/systemd/system/cdp-reopen-luks-volume.service [hunk 1]
```diff
@@ -14,7 +14,7 @@ StartLimitBurst=5
 
 [Service]
 Type=oneshot
-ExecStart=/etc/cdp-luks/bin/reopen-luks-volume.sh
+ExecStart=/bin/bash -c '/etc/cdp-luks/bin/reopen-luks-volume.sh | tee /var/log/cdp-luks/reopen-luks-volume-$(date +%s).log'
```
[the remark this answers is not in the document]
Comment (gregito) [reply saltstack/base/salt/luks/etc/systemd/system/cdp-reopen-luks-volume.service]: thank you! :)

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM

File: saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh [hunk 2]
```diff
@@ -82,13 +82,13 @@ create_luks_volume() {
...
-  /etc/cdp-luks/bin/create-luks-volume.sh
+  /etc/cdp-luks/bin/create-luks-volume.sh | tee /var/log/cdp-luks/create-luks-volume.log
   echo "LUKS volume creation finished."
 }
 
 populate_luks_volume() {
   echo "LUKS volume population with secrets started."
-  /etc/cdp-luks/bin/populate-luks-volume.sh
+  /etc/cdp-luks/bin/populate-luks-volume.sh | tee /var/log/cdp-luks/populate-luks-volume.log
   echo "LUKS volume population with secrets finished."
```
Comment (lajosrodek) [inline saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh:92]: Sorry, forgot mentioning that we should think about the `${PIPESTATUS[0]}` thing in these script executions as well. There are two choices: either `exit` explicitly abort if `${PIPESTATUS[0]}` is nonzero, or just `return` from the function with the nonzero `${PIPESTATUS[0]}`. As I see, https://github.infra.cloudera.com/cloudbreak/cloudbreak/pull/2733 / [CB-27096](https://jira.cloudera.com/browse/CB-27096) will force `user-data-helper.sh` abort if any command execution fails, so even the `return` option is satisfactory.

**Review by lajosrodek — CHANGES_REQUESTED** [review]
Sorry, a few more asks.

**Review by TheTinkerDad — APPROVED** [review]

**Pull request merged**